### PR TITLE
Various data-binding fixes.

### LINKED ISSUE #30: Children added to <iron-selector> before it is added to document not marked selected
If I programatically construct an `<iron-selector>` element like this ([jsbin, slow to load](http://jsbin.com/qajela/2/edit?html,output)):
```javascript
    var parent = document.createElement('iron-selector');
    parent.selected = "0";
    
    var child = document.createElement('div');
    child.innerHTML = 'Item 0';
    
    Polymer.dom(parent).appendChild(child);
    document.body.appendChild(parent);
```
then the child `<div>` does not receive the `iron-selected` class I hoped for (using Chrome). Possibly this is considered unsupported as I set the `selected` property before appending the corresponding child; but if I simply reverse the final two lines - adding parent to document before adding child to parent - the class is set correctly ([jsbin](http://jsbin.com/qajela/1/edit?html,output)).

It looks like the inconsistency is due to [iron-selectable.html](https://github.com/PolymerElements/iron-selector/blob/master/iron-selectable.html) only setting up its `MutationObservers` during the `attach()` callback. Naively it might be fixed by either calling `_updateSelected()` directly during `attach()` ([diff](https://github.com/PolymerElements/iron-selector/compare/master...shaydon:update-on-attach)) or by setting up the observers during `ready()` ([diff](https://github.com/PolymerElements/iron-selector/compare/master...shaydon:observe-earlier)), though I don't know enough about Polymer to know if either of these would be problematic.

My reason for wanting the quoted code to work is simply that I'm exploring using Polymer elements with a library (https://github.com/Matt-Esch/virtual-dom) that happens to construct elements in the order element -> properties -> children. All else being equal it does seem desirable that an element's state be a pure function of its public properties.

### LINKED ISSUE #33: Bindings to `selectedItems` do not notify interested parties
repo: http://jsbin.com/guniqopiwi/1/edit

For example, `_computeTitle()` only gets called after the first selection is made. Subsequent taps on items do not call it:

```
<template id="t" is="dom-bind">
  <iron-selector multi
                 selected-attribute="selected"
                 selected-items="{{selectedThreads}}">
    <template is="dom-repeat" items="[[threads]]">
      <div>{{item.name}}</div>
    </template>
  </iron-selector>
  How many selected: <span>{{_computeTitle(selectedThreads.length)}}</span>
</template>
```

### LINKED ISSUE #43: Impossible to set selected-values on elements created with dom-repeat
```html
<paper-menu multi selected-values="[0]">
  <template is="dom-repeat" items="{{items}}">
    <paper-item>
      <span>{{item.name}}</span>
    </paper-item>
  </template>
  <paper-item>
    <span>test</span>
  </paper-item>
</paper-menu>
```

For example that code select the 'test' item

### LINKED ISSUE #44: multi selection's on-selected-values-changed will not invoked when it changes
I found the source will assigned array directly: 

 - https://github.com/PolymerElements/iron-selector/blob/master/iron-multi-selectable.html#L106
 - https://github.com/PolymerElements/iron-selector/blob/master/iron-multi-selectable.html#L108

and I think use Polymer's array operation: `this.push('selectedValues', value)`, `this.splice('selectedValues', i, 1)` will solve this problem.

### PR DESCRIPTION
**Pull request by cdata**
Fixes #30
Fixes #33
Fixes #43
Fixes #44

Credit to Brandon Myers <trakkasure@gmail.com, @Trakkasure> for doing
the groundwork for these fixes.

/cc @frankiefu

### CODE REVIEW

File: iron-multi-selectable.html [hunk 1]
```diff
@@ -72,6 +72,11 @@
       this._selection.multi = multi;
     },
 
+    get _shouldUpdateSelected() {
+      return (this.multi && this.selectedValues && this.selectedValues.length) ||
+        Polymer.IronSelectableBehavior._shouldUpdateSelected;
+    },
```
Comment (frankiefu) [inline iron-multi-selectable.html]: I don't see anywhere in `iron-selectable` or `iron-multi-selectable` using `_shouldUpdateSelected`.  Is it for something in a separate changes?

File: iron-multi-selectable.html [hunk 2]
```diff
@@ -103,9 +103,9 @@
       var i = this.selectedValues.indexOf(value);
       var unselected = i < 0;
       if (unselected) {
-        this.selectedValues.push(value);
+        this.push('selectedValues',value);
```
Comment (notwaldorf) [inline iron-multi-selectable.html]: 😳bane of my existence

File: test/multi.html [hunk 3]
```diff
@@ -43,6 +44,20 @@
     </template>
   </test-fixture>
 
+  <!--
+  NOTE(cdata): Enable test-fixture when polymer/polymer#2495 is resolved
+  -->
+  <!--<test-fixture id="repeatedItems">
```
Comment (notwaldorf) [inline test/multi.html]: Theory: enable the test, but expect it to fail always in Safari? WDYT?
(at least then we'd get slighty more coverage)

**Comment by notwaldorf** [comment]
minor comments, otherwise this looks 👌

File: test/multi.html (hunk 3 above)
Comment (cdata) [inline test/multi.html]: Right now, the test runs. The `test-fixture` was borked, so there is a danger that if we use these elements in other tests, there will be state leak. Also: the problem in the cited issue affects all IE.

Comment (notwaldorf) [inline test/multi.html]: Ohhh i misread what's getting commented out. Yeaaaaah, :+1:

**Pull request merged**